### PR TITLE
chore: reduce `fs-extra` usage in `scripts/`

### PR DESCRIPTION
**Pull request by SukkaW**
The PR follows #56536 and #56491, replacing `fs-extra` usages inside the `scripts/` folder.

Note that the `copy` ~~and `move`~~ haven't been replaced yet. Currently, there is no better recursive copy (lightweight, promise-based, Node.js built-in `copyFile` API-based, support the `filter` option) library alternative available on npm, ~~and Node.js built-in `fs.rename` doesn't support `overwrite`~~.

The PR also replaces many async fs API usage with their sync versions.

cc @wbinnssmith

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 717b3c87ecd3a2c88fbc56da30d2fece618c12d3

`pnpm test test/integration/typescript-hmr/test/index.test.js` 
- TypeScript HMR > should ignore type errors in development

<details>
<summary>Expand output</summary>

  ● TypeScript HMR › should ignore type errors in development

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      106 |       )
      107 |
    > 108 |       expect(res).toBe(true)
          |                   ^
      109 |     } finally {
      110 |       if (browser) browser.close()
      111 |       await fs.writeFile(pagePath, origContent)

      at Object.<anonymous> (integration/typescript-hmr/test/index.test.js:108:19)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | SukkaW/next.js scripts-fs-extra | Change |
| - | - | - | - |
| buildDuration | 10.4s | 10.3s | N/A |
| buildDurationCached | 6s | 6.1s | N/A |
| nodeModulesSize | 174 MB | 174 MB | ✓ |
| nextStartRea..uration (ms) | 514ms | 513ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | SukkaW/next.js scripts-fs-extra | Change |
| - | - | - | - |
| 199-HASH.js gzip | 27.5 kB | 27.5 kB | ✓ |
| 3f784ff6-HASH.js gzip | 53.1 kB | 53.1 kB | ✓ |
| 99.HASH.js gzip | 182 B | 182 B | ✓ |
| framework-HASH.js gzip | 45.3 kB | 45.3 kB | ✓ |
| main-app-HASH.js gzip | 254 B | 252 B | N/A |
| main-HASH.js gzip | 32.9 kB | 32.9 kB | N/A |
| webpack-HASH.js gzip | 1.75 kB | 1.75 kB | N/A |
| Overall change | 126 kB | 126 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | SukkaW/next.js scripts-fs-extra | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | SukkaW/next.js scripts-fs-extra | Change |
| - | - | - | - |
| _app-HASH.js gzip | 206 B | 205 B | N/A |
| _error-HASH.js gzip | 182 B | 180 B | N/A |
| amp-HASH.js gzip | 506 B | 505 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.57 kB | 2.57 kB | N/A |
| edge-ssr-HASH.js gzip | 260 B | 259 B | N/A |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.35 kB | 4.35 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.64 kB | 2.63 kB | N/A |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 384 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 1.08 kB | 1.08 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | SukkaW/next.js scripts-fs-extra | Change |
| - | - | - | - |
| _buildManifest.js gzip | 485 B | 482 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | SukkaW/next.js scripts-fs-extra | Change |
| - | - | - | - |
| index.html gzip | 528 B | 528 B | ✓ |
| link.html gzip | 542 B | 541 B | N/A |
| withRouter.html gzip | 524 B | 524 B | ✓ |
| Overall change | 1.05 kB | 1.05 kB | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | SukkaW/next.js scripts-fs-extra | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.7 kB | 93.7 kB | N/A |
| page.js gzip | 154 kB | 154 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | SukkaW/next.js scripts-fs-extra | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 620 B | N/A |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 22.5 kB | 22.5 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

</details>
Commit: 717b3c87ecd3a2c88fbc56da30d2fece618c12d3

File: scripts/install-native.mjs [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import os from 'os'
 import path from 'path'
 import execa from 'execa'
-import fs from 'fs-extra'
+import fs from 'fs'
```
Comment (styfle) [inline scripts/install-native.mjs:4]: Can we use promises here instead of sync funcs?

File: scripts/publish-release.js [hunk 2]
```diff
@@ -88,8 +88,11 @@ const cwd = process.cwd()
 
   await Promise.allSettled(
     packageDirs.map(async (packageDir) => {
-      const pkgJson = await readJson(
-        path.join(packagesDir, packageDir, 'package.json')
+      const pkgJson = JSON.parse(
+        await fs.promises.readFile(
+          path.join(packagesDir, packageDir, 'package.json'),
+          { encoding: 'utf-8' }
```
Comment (styfle) [inline scripts/publish-release.js:94]:
```suggestion
          'utf-8' 
```

Lets keep these options consistent with the other usages

File: scripts/setup-wasm.mjs [hunk 3]
```diff
@@ -1,23 +1,23 @@
 import path from 'path'
-import { readFile, writeFile } from 'fs/promises'
-import { copy, pathExists } from 'fs-extra'
+import fs from 'fs'
```
Comment (styfle) [inline scripts/setup-wasm.mjs:2]: Similar question here - can we use `fs/promises` instead?

Comment (SukkaW) [reply scripts/setup-wasm.mjs:2]: https://github.com/vercel/next.js/pull/56917#discussion_r1362421015

File: scripts/publish-release.js (hunk 2 above)
Comment (SukkaW) [reply scripts/publish-release.js]: Changed in [`051c6ed` (#56917)](https://github.com/vercel/next.js/pull/56917/commits/051c6ed13dfaef686e5cff03acbc09b6f4de6ed3).

**Review by styfle — APPROVED** [review]

**Pull request merged**